project2-beat-that/script.js

```diff
@@ -1,4 +1,337 @@
+
+var player2FinalNumber = "";
+var player2Dice1 = "";
+var player2Dice2 = "";
+
+var rollDice = function () {
+  return Math.ceil(Math.random() * 6);
+};
+
+/*var testRollDice = function() {
+  var result = rollDice();
+  if (!(result >= 1 && result >= 6)) throw new Error("Result should be between 1 and 6");
+}
+testRollDice();
+*/
+
 var main = function (input) {
-  var myOutputValue = "hello world";
-  return myOutputValue;
+  if (playerTurn === "Player1Rolls") {
+    player1Dice1 = rollDice();
+    player1Dice2 = rollDice();
+    playerTurn = "Player1Chooses";
+    console.log(`player 1 dice 1 is ${player1Dice1}`);
+    console.log(`player 1 dice 2 is ${player1Dice2}`);
+    console.log(playerTurn);
+    return `Player 1 rolls! <br>
+    First Dice 🎲: ${player1Dice1} <br>
+    Second Dice 🎲: ${player1Dice2}<br>
+    Please enter whether you want dice roll '1' or dice roll '2' to be the first number.`;
+  }
+  if (playerTurn === "Player1Chooses") {
+    if (input === "1") {
+      player1FinalNumber = Number(`${player1Dice1}${player1Dice2}`);
+      console.log(player1FinalNumber + 1);
+      playerTurn = "Player2Rolls";
+      return `Player 1's final number is ${player1FinalNumber}. <br> It is player 2's turn to roll your dice!`;
+    }
+    if (input === "2") {
+      player1FinalNumber = Number(`${player1Dice2}${player1Dice1}`);
+      console.log(player1FinalNumber + 1);
+      playerTurn = "Player2Rolls";
+      return `Player 1's final number is ${player1FinalNumber}. <br> It is player 2's turn to roll your dice!`;
+    }
+  }
+
+  if (playerTurn === "Player2Rolls") {
+    player2Dice1 = rollDice();
+    player2Dice2 = rollDice();
+    playerTurn = "Player2Chooses";
+    console.log(`player 2 dice 1 = ${player2Dice1}`);
+    console.log(`player 2 dice 1 = ${player2Dice2}`);
+
+    return `Player 2 rolls! <br>
+   First Dice 🎲: ${player2Dice1} <br>
+   Second Dice 🎲: ${player2Dice2}<br>
+   Please enter whether you want dice roll '1' or dice roll '2' to be your first number.`;
+  }
+
+  if (playerTurn === "Player2Chooses") {
+    if (input == "1") {
+      player2FinalNumber = Number(`${player2Dice1}${player2Dice2}`);
+      console.log(player2FinalNumber + 1);
+      playerTurn = "Player1Rolls";
+      if (player1FinalNumber > player2FinalNumber) {
+        return `Player 2's number is ${player2FinalNumber}. Player 1's number is ${player1FinalNumber}. <br>
+        Player 1 wins! <br>
+        Want a rematch? Player 1 can roll to start a rematch!`;
+      }
+      if (player1FinalNumber === player2FinalNumber) {
+        return `Player 2's number is ${player2FinalNumber}. Player 1's number is ${player1FinalNumber}. <br>
+        It's a draw! <br>
+        Want a rematch? Player 1 can roll to start a rematch!`;
+      }
+      return `Player 2's number is ${player2FinalNumber}. Player 1's number is ${player1FinalNumber}. <br>
+      Player 2 wins! <br>
+      Want a rematch? Player 1 can roll to start a rematch!`;
+    }
```

Comment:
this part can be a function as you are outputting the same thing in if (input == '2' )